### DISCUSSION TITLE
How to ensure renovate groups monorepo dependencies into 1 PR

### DISCUSSION THREAD

**Discussion by EvHaus** [renovatebot/renovate#18164] [Request Help] [answered]
Hi there,

I'm the maintainer of [eslint-config-globex](https://github.com/GlobexDesignsInc/eslint-config-globex) which is a monorepo that publishes several npm packages. How can I ensure that when some projects uses more than 1 of the exported packages from my repo, renovate bot will group them into 1 PR -- like it does for many other projects ([example](https://github.com/EvHaus/haus.gg/pull/427))?

Right now, it creates separate PRs for each package (examples [1](https://github.com/EvHaus/haus.gg/pull/470), [2](https://github.com/EvHaus/haus.gg/pull/469)).

I can't figure why this happens. How does renovate determine if some npm packages come from a single source?

**Comment by tn185070** [comment, marked as answer]
Hi, not a maintainer, but possibly https://docs.renovatebot.com/configuration-options/#packagerules answers your question?

**Reply by EvHaus** [reply]
From what I understand, `packageRules` is what the users of my library would need to set to group my packages. How can I -- as the author of the package -- ensure that my users get this behaviour by default? From what I can see, `react`, `babel`, `next` and other popular monorepo projects do not need to set any of this and yet Renovate knows to automatically group those packages together.

**Reply by tn185070** [reply]
https://docs.renovatebot.com/config-presets/#contributing-to-presets?

**Reply by EvHaus** [reply]
Aah, I think I understand... Are you saying those monorepos are [hardcoded into Renovate itself](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/packages.ts)?

**Reply by tn185070** [reply]
I've just been reading the docs a lot today while trying to troubleshoot an issue, I'll wait for a maintainer to answer this :D

**Reply by viceice** [reply]
yes, they are hardcoded

https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/monorepo.ts#L6

**Reply by EvHaus** [reply]
I see. And I'm guessing my package is not popular enough to be considered for the list?

**Reply by viceice** [reply]
feel free to open a PR to add your packages

**Reply by EvHaus** [reply]
Done! https://github.com/renovatebot/renovate/pull/18169 Thanks for the consideration.
